indexing-hadoop/src/main/java/io/druid/indexer/HadoopDruidIndexerConfig.java

```diff
@@ -115,6 +118,13 @@ public void configure(Binder binder)
     INDEX_IO = injector.getInstance(IndexIO.class);
     INDEX_MERGER = injector.getInstance(IndexMerger.class);
     INDEX_MERGER_V9 = injector.getInstance(IndexMergerV9.class);
+    LIFECYCLE = injector.getInstance(Lifecycle.class);
+    try {
+      LIFECYCLE.start();
```

Comment:
@gianm let me explain first how the auth will work.
The first way is the case where user is using HDFS storage module, in that case the user has to provide some properties `druid.hdfs.security.kerberos.keytab/principal` then the hdfs module will do the call to the login function to set some static hadoop static classes and some token. So in theory one call per JVM will be enough. No the issue with this is the fact that when we switch classloader, all the stuff done at the peon start is lost thus we need to call again lifecyle.start to initiate that hook in the HDFS module.

The second way is as you said the user is not using HDFS storage module, then in that case with the hadoop task user must provide the 2 fields (principal/keytab) under tuning config  and will use the Jobhelper.authenicate just before we call ensurePath method. 

As you see it will be nice if the user is using HDFS module he doens't need to use method 2. 
Hope this is clear.

Now the bigger question don't u think that it is a bug that the lifecycle doesn't start under the new class loader ? if i develop a module to work with hadoop i expect that i can rely on the management of lifecycle right ?